### DISCUSSION TITLE
Having issues downloading v0.11.8 - Virus Detected

### DISCUSSION THREAD

**Discussion by ka3yan** [g4ixt/QtTinySA#69] [Q&A] [answered]
I'm having an issue with downloading the executable for v0.11.8.  Windows detects a virus.  I'm sure there isn't one, but obviously I don't want to disable Windows virus protection.  Just curious if something has changed that's causing this.  I'm on Windows 11 Pro 23H2 build 22631.4317.

**Comment by g4ixt** [comment, marked as answer]
Hi ka3yan,

There isn't anything I can do about that.

When I compiled the programme using Nuitka on my Win-11 machine on 1st October, Windows Defender didn't pop up with any issues but when I run it now, it does.  I have checked that the 'checksums' on the GitHub file are the same as my original.  I have tried uploading the executable to 'virustotal' and it throws up a lot of positives for different things - if you do a search on Google you will see that this is a common issue with compiled Python programmes.

There is supposed to be a way to report false positives to Microsoft for analysis, but when I try that it says "page doesn't exist".

In conclusion it's up to you whether you allow it to be installed or not but the code is public and the Nuitka compiler is public.  If you want to be sure you are not exposed to a virus then run it in Linux or use the Python version, for which you can read the code and check what is in it.

Ian

**Comment by ka3yan** [comment]
I’m curious if you package as a .zip instead of a .exe if we’d have the same issue?  It might just be that Windows doesn’t like it being an executable.

 

Brad

**Reply by g4ixt** [reply]
I doubt if that would make any difference or people would be hiding viruses in zip files

**Reply by ka3yan** [reply]
I know this is now getting a little long in the tooth, but for reference, Windows Defender thinks the file contains the trojan horse: Ulthar.A!ml

Since the executable isn't something that I can say with confidence doesn't contain that trojan horse, I'll leave it alone.  Just wanted to provide as many details as I have.

Brad

**Reply by g4ixt** [reply]
Hi Brad,

Thanks for letting me know.  I used to build Windows executables using 'pyinstaller' before I started to move to the 'nuitka' compiler, so I will fire up my Windows system and try one like that again and attach it here for you to try.

73 de Ian

**Comment by ka3yan** [comment]
Thanks Ian. I appreciate it and the work you’ve done. 73 and Merry Christmas,BradSent from my iPhoneOn Dec 24, 2024, at 04:41, Ian ***@***.***> wrote:﻿
Hi Brad,
Thanks for letting me know.  I used to build Windows executables using 'pyinstaller' before I started to move to the 'nuitka' compiler, so I will fire up my Windows system and try one like that again and attach it here for you to try.
73 de Ian

**Reply by g4ixt** [reply]
I have added "QtTinySA_W11_by_pyinstaller.exe" in the releases section so please have a try of that.  Windows defender quarantined the Nuitka one on my system (despite it not doing that when I created it).  I will update my Nuitka compiler to see if that makes any difference, the version I have is from October.

**Comment by ka3yan** [comment]
Thanks Ian. As you’d expect, tomorrow I’ll be a bit preoccupied, but as soon as I can I’ll give it a try. Thank you!!BradSent from my iPhoneOn Dec 24, 2024, at 12:29, Ian ***@***.***> wrote:﻿
I have added "QtTinySA_W11_by_pyinstaller.exe" in the releases section so please have a try of that.  Windows defender quarantined the Nuitka one on my system (despite it not doing that when I created it).  I will update my Nuitka compiler to see if that makes any difference, the version I have is from October.

**Comment by ka3yan** [comment]
Ian,

 

It doesn’t look like the PyInstaller executable works (see attached screenshot).  I was getting a popup terminal window that opens and immediately closes, so I ran it in the CMD prompt.  I just wanted to let you know that it wouldn’t install here.

 

Brad

**Reply by g4ixt** [reply]
No screenshot attached?

**Comment by ka3yan** [comment]
Let’s see if you see it now?

**Reply by g4ixt** [reply]
No.  Am I looking in the wrong place?  I can't see where in this discussion it would be?

**Comment by Richard-G4WKW** [comment]
I am running Windows 10 Pro Version 22H2 (OS Builtd 19045.5247), as reported by winver.  I am getting nowhere fast trying to use v0.11.8 software:

1. If I attempt to run QtTinySA.exe from an Administrator command prompt it comes back with "Access Denied" and pops up a blue back ground dialog box with "This app can't run on your PC" and "To find a version for your PC, check with the software publisher."

2. If I attempt to run QtTinySA_W11_by_pyinstaller.exe from an Administrator command prompt it comes back with:

D:\test>QtTinySA_W11_by_pyinstaller.exe
Traceback (most recent call last):
  File "QtTinySA.py", line 43, in <module>
ModuleNotFoundError: No module named 'QtTinySpectrum'
[23928] Failed to execute script 'QtTinySA' due to unhandled exception!

Hope that you can resolve this!

**Reply by g4ixt** [reply]
I will have a look.

**Comment by g4ixt** [comment]
Hello Brad & Richard,

apologies for the delay in looking at this again, I have been busy and I rarely boot my PC into Windows.  I have now found two things:
 
1. I had forgotten to reset my system to the 'main' branch before compiling with 'pyinstaller', so had unintentionally built my development branch v0.11.9 which is 'work in progress' and not functional (although it does open in Python).
2. The Windows .exe doesn't work for me either, with exactly the same behaviour that Richard sees.
 
I think on Christmas Eve after I built the pyinstaller executable I must have then run the wrong one as I thought it worked.
Sorry about that, I will update things and have another go.

I still can't see any screenshots, by the way, where should I be looking?
 
Ian

**Reply by Richard-G4WKW** [reply]
Hi Ian,

Many thanks for your reply.

> I still can't see any screenshots, by the way, where should I be looking?

Well, I did not try and include any screenshots as I thought the text I copied from the command prompt should give you a good idea of the problem experienced when attempting to use QtTinySA_W11_by_pyinstaller.exe.

At the moment, I am successfully using the tinySA-App.exe to connect, after updating the firmware.  I would be pleased to try the QtTinySA version too.

tinySA-ULTRRA ZS-406
Version: tinySA4_v1.4-192-g73fe677
Build Time: Dec 23 2024 - 13:25:46

Kind regards,

Richard.

**Reply by g4ixt** [reply]
Richard,

The comment about the screenshot was for Brad. 

I have just added a file to the "[v0.11.8 Releases](https://github.com/g4ixt/QtTinySA/releases/tag/v0.11.8)" called QtTinySA.zip which includes both a Windows executable (built on 07.01.2025 using pyinstaller) and the config database file.  If you extract them both to a folder and run the .exe it should copy the database to a personal folder when you run it.

On my system, Windows Defender doesn't flag the .exe as problematic.  I'd be interested to know how you find QtTinySA versus tinySA-App and if there is anything from the other App you think I could add.  I don't plan on adding a firmware update feature though as the dfu-util method works perfectly every time for me.

Ian

**Comment by ka3yan** [comment]
Ian,

 

Heyyyyy!!  It works!! I was able to download and run the executable from the unzipped folder.  No issues at all.  Thanks for fixing this!!

 

Brad
